pymatgen/core/structure.py

```diff
@@ -764,6 +764,28 @@ def get_neighbors(self, site, r, include_index=False):
                                       include_index=include_index)
         return [d for d in nn if site != d[0]]
 
+    def get_nearest_site(self, coords, site, r = None):
```

Comment:
If this is only used by ferroelectric analysis for now, I suggest you do not add this as a core function. I am pondering how many use cases there are for this function.